src/Microsoft.AspNet.OData.Shared/Formatter/Deserialization/ODataResourceDeserializer.cs

```diff
@@ -235,11 +235,11 @@ public virtual object CreateResourceInstance(IEdmStructuredTypeReference structu
                             structuredType.StructuredDefinition(), model);
 
                         return Activator.CreateInstance(readContext.ResourceType, clrType, structuralProperties,
-                            dynamicDictionaryPropertyInfo);
+                            dynamicDictionaryPropertyInfo, structuredType.IsComplex());
```

Comment:
>, structuredType.IsComplex()) [](http://example.com/codeflow?start=57&length=29)

are there other places where we should be setting this?

Reply:
Not really. Ideally all resources are created here.